homeassistant/components/mqtt/client.py

```diff
@@ -663,7 +673,13 @@ def _mqtt_handle_message(self, msg: MQTTMessage) -> None:
             )
         self._mqtt_data.state_write_requests.process_write_state_requests()
 
-    def _mqtt_on_callback(self, _mqttc, _userdata, mid, _granted_qos=None) -> None:
+    def _mqtt_on_callback(
+        self,
+        _mqttc: mqtt.Client,
+        _userdata: None,
+        mid: int,
+        _granted_qos: tuple[Any] | None = None,
```

Comment:
```suggestion
        _granted_qos: tuple[Any, ...] | None = None,
```

Just `tuple[Any]` would only allow for a single element.